store/sqlstore/post_store.go

```diff
@@ -1231,7 +1231,8 @@ func (s *SqlPostStore) getPostsSinceCollapsedThreads(options model.GetPostsSince
 		Where(sq.Gt{"Posts.UpdateAt": options.Time}).
 		Where(sq.Eq{"Posts.RootId": ""}).
 		OrderBy("Posts.CreateAt DESC").
-		Limit(1000).
+		Offset(uint64(options.Page * options.PerPage)).
+		Limit(uint64(options.PerPage)).
```

Comment:
if perPage is 0, should we keep the limit of `1000` ?

Reply:
It's already added here - https://github.com/mattermost/mattermost-server/pull/21026/files/1d1d6ee72eb7ee052acd94ae72f22659ad3332e4#diff-17a4db18ae781c89b1357c12bb74c8f78396103865f014b1e2c413e4739e7dc1R1257